tests/kerastuner/engine/tuner_correctness_test.py

```diff
@@ -272,3 +272,7 @@ def on_train_begin(self, logs):
                  callbacks=[logging_callback])
 
     assert len(callback_instances) == 6
+
+
+def test_on_train_begin_existence(tmp_dir):
```

Comment:
I think we should test that this method is actually called in `run_trial`, i.e. something like:

```
class MyTuner(Tuner):
  def _on_train_begin(self, ...):
    self.was_called = True

tuner = MyTuner()
tuner.run_trial(...)
assert tuner.was_called
```